datasources/gab/search_gab.py

```diff
@@ -0,0 +1,83 @@
+"""
+Import scraped Gab data
+"""
+import datetime
+import re
+
+from backend.lib.search import Search
+
+
+class SearchGab(Search):
+    """
+    Import scraped gab data
+    """
+    type = "gab-search"  # job ID
+    category = "Search"  # category
+    title = "Import scraped Gab data"  # title displayed in UI
+    description = "Import Gab data collected with an external tool such as Zeeschuimer."  # description displayed in UI
+    extension = "ndjson"  # extension of result file, used internally and in UI
+    is_from_extension = True
+    fake = ""
+
+    # not available as a processor for existing datasets
+    accepts = [None]
+
+    def get_items(self, query):
+        """
+        Run custom search
+
+        Not available for Gab
+        """
+        raise NotImplementedError("Gab datasets can only be created by importing data from elsewhere")
+
+    @staticmethod
+    def map_item(post):
+        """
+        Parse Gab post
+
+        :param node:  Data as received from Gab
+        :return dict:  Mapped item
+        """
+        
+        post_time = datetime.datetime.strptime(post["ca"], "%Y-%m-%dT%H:%M:%S.%fZ")
+        mapped_item = {
+            "id": post["i"],
+            "created_at": post["ca"],
+            "body": post["c"],
+            "url": post["ul"],
+            "reaction_count": post["fc"] if post["fc"] else 0,
```

Comment:
Running some tests and not all items have `fc` (and thus 4CAT cannot display them or use them for analysis as `post["fc"]` results in an error). I am not very familiar with Gab, but it does look like if there is no `fc`, there are no reactions. If that is the case `post.get("fc", 0)` ought to be fine.

Reply:
got it!